### PR TITLE
Setting up compression using flask-compress

### PR DESCRIPTION
**Pull request by mistercrunch**

### CODE REVIEW

File: superset/utils.py [hunk 1]
```diff
@@ -838,3 +843,38 @@ def get_filter_key(f):
 def get_update_perms_flag():
     val = os.environ.get('SUPERSET_UPDATE_PERMS')
     return val.lower() not in ('0', 'false', 'no') if val else True
+
+
+def gzipped(f):
```
Comment (john-bodley) [inline superset/utils.py]: Max is this required? I felt that `Flask-Compress` takes care of all of this. Also would it be possible to report the time it takes to execute the `sql_json` and `explore_json` endpoints w/ and w/o caching?

Comment (mistercrunch) [reply superset/utils.py]: Flask-Compress only takes care of static files :( , I agree that this seems in-scope for it, I'll open a PR with them in parallel to this. 

I actually tested by looking at headers and the decorator does do the difference.

Looks like it's not py3 compatible, going to tweak this now.

Comment (mistercrunch) [reply superset/utils.py]: Actually ran more tests and you're right, I think the payload I used for testing didn't meet flask-compress threshold for zipping on my first round of tests...

**Comment by hughhhh** [comment]
🚢

**Comment by john-bodley** [comment]
LGTM

**Comment by maver1ck** [comment]
How to disable this option ?
I have nginx in front of Superset for ssl and compression.

**Comment by mistercrunch** [comment]
@maver1ck oops, we should make that optional

**Comment by mistercrunch** [comment]
@maver1ck https://github.com/apache/incubator-superset/pull/4617

**Pull request merged**